core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -1490,4 +1490,20 @@ object TestUtils extends Logging {
       Metrics.defaultRegistry.removeMetric(metricName)
   }
 
+  def stringifyTopicPartitions(partitions: Set[TopicPartition]): String = {
+    Json.legacyEncodeAsString(
+      Map(
+        "version" -> 1,
+        "partitions" -> partitions.map(tp => Map("topic" -> tp.topic, "partition" -> tp.partition))
+      )
+    )
+  }
+
+  def resource[R <: AutoCloseable, A](resource: R)(func: R => A): A = {
```

Comment:
This seems similar to try-with-resources. Does Scala have an equivalent construct?

Reply:
Yep. For Scala 2.13 they are going to add it to the standard library. Until that is release and we upgrade to it we need to implement it ourselves.